### PR TITLE
Add Bundler detection and change message from update_checker

### PR DESCRIPTION
**Pull request by jlnquere**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:
- [x] Run `rspec` for all tools you modified
- [x] Run `rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

Here is the code for the Issue #6560

It simply add a `is_bundler` is `Helper` and change the displayed message in `update_checker`.

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 1]
```diff
@@ -47,6 +56,10 @@ def self.is_ci?
       ci?
     end
 
+    def self.is_bundler?
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/helper.rb]: We don't actually need this method, we only have some of those duplicates to support legacy code bases. Let's remove this and stick with `bundler?`

File: fastlane_core/lib/fastlane_core/update_checker/update_checker.rb [hunk 2]
```diff
@@ -53,9 +53,14 @@ def self.show_update_message(gem_name, current_version)
       puts '#######################################################################'.green
       puts "# #{gem_name} #{available} is available. You are on #{current_version}.".green
       puts "# It is recommended to use the latest version.".green
-      puts "# Update using 'sudo gem update #{gem_name.downcase}'.".green
+      if Helper.is_bundler?
+        puts "# Update using 'bundle update #{gem_name.downcase}'.".green
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/update_checker/update_checker.rb]: 😍

File: fastlane_core/spec/helper_spec.rb [hunk 3]
```diff
@@ -1,5 +1,22 @@
 describe FastlaneCore do
   describe FastlaneCore::Helper do
+    describe "#is_bundler?" do
+      it "returns false when not in a bundler environment" do
+        stub_const('ENV', {})
+        expect(FastlaneCore::Helper.is_bundler?).to be false
```
Comment (KrauseFx) [inline fastlane_core/spec/helper_spec.rb]: Let's  test `bundler?` directly here

File: fastlane_core/lib/fastlane_core/update_checker/update_checker.rb [hunk 4]
```diff
@@ -53,9 +53,14 @@ def self.show_update_message(gem_name, current_version)
...
+      else
+        puts "# Update using 'sudo gem update #{gem_name.downcase}'.".green
+      end
       puts "# To see what's new, open https://github.com/fastlane/#{gem_name}/releases.".green if ENV["FASTLANE_HIDE_CHANGELOG"]
-      if Random.rand(5) == 1
+
+      if Random.rand(5) == 1 && !Helper.is_bundler?
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/update_checker/update_checker.rb]: Let's use `bundler?`

File: fastlane_core/lib/fastlane_core/update_checker/update_checker.rb [hunk 5]
```diff
@@ -53,9 +53,14 @@ def self.show_update_message(gem_name, current_version)
       puts '#######################################################################'.green
       puts "# #{gem_name} #{available} is available. You are on #{current_version}.".green
       puts "# It is recommended to use the latest version.".green
-      puts "# Update using 'sudo gem update #{gem_name.downcase}'.".green
+      if Helper.is_bundler?
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/update_checker/update_checker.rb]: Let's use `bundler?`

**Review by KrauseFx — CHANGES_REQUESTED** [review]

**Comment by KrauseFx** [comment]
This is a great addition, thanks @Onejjy 👍  I added some smaller comments before it's ready to be merged 🚀

**Comment by Onejjy** [comment]
@KrauseFx all requested changes have been made :)

Rubocop report 1 offense on the code:

```
lib/fastlane_core/helper.rb:5:3: C: Module has too many lines. [106/100]
  module Helper ...
  ^^^^^^^^^^^^^

```

I don't really see how to get rid of that. Maybe by splitting into a new module `All Xcode Related things` (from `xcode_path` to `itms_path`) ?

**Comment by KrauseFx** [comment]
You can add the following right above `module Helper`
```ruby
module FastlaneCore
  # rubocop:disable Metrics/ModuleLength
  module Helper
    ...
```

**Comment by KrauseFx** [comment]
Thanks for updating the PR 👍

**Comment by KrauseFx** [comment]
I updated the PR with the new rubocop rules, let's see if the build is happy now 👍

**Comment by Onejjy** [comment]
Rubocop is happy now ! Thank you for the tip :)

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
This is great, thanks again for your contribution, this will be in the next release  🚀

**Pull request merged**